examples/howto/plot_simulate_mpi_backend.py

```diff
@@ -47,8 +47,11 @@
 # ``openmpi``, which must be installed on the system
 from hnn_core import MPIBackend
 
-with MPIBackend(n_procs=2, mpi_cmd='mpiexec'):
-    dpls = simulate_dipole(net, tstop=310., n_trials=1)
+# to create a parent MPI job that uses MPIBackend to spawn child jobs, set
+# mpi_comm_spawn=True and call this script from terminal using
+# ``$ mpiexec -np 1 --oversubscribe python -m mpi4py /path/to/hnn-core/examples/howto/plot_simulate_mpi_backend.py``
```

Comment:
so the MPI spawning jobs only works if MPI is calling Python?

Reply:
MPI always calls Python, which was previously only accomplished by having [`suprocess.Popen`](https://docs.python.org/3/library/subprocess.html) call [`mpiexec`](https://github.com/jonescompneurolab/hnn-core/blob/c7460ba27cfc95343c83a41f2415f510ebf67fcd/hnn_core/parallel_backends.py#L648) to start multiple Python processes. Now, `mpiexec` can be called explicitly by the user or in a Slurm sbatch file.

Reply:
I suppose I'm wondering why MPI's API is not completely through Python rather than partially command-line and partially Python. How would you debug an error? Normally in Ipython, you can execute parts of the code for debugging or have breakpoints ...

see also: https://ipython.org/ipython-doc/3/parallel/parallel_mpi.html

Reply:
Error debugging can be tricky with MPI, but that is why mpi4py gives you access to all of the MPI communicator objects, etc. Re: parallel processing in ipython, even their parallel engines require a call to `mpiexec`. They appear to offer the ability for advanced users to call `mpiexec` themselves, or for users to us an automated system-level function (`ipcluster start`) that internally calls `mpiexec`.